File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int mini = 1e9;

    void func(TreeNode* root, TreeNode*& prev){
        if(root->left!=NULL) func(root->left, prev);
        if(prev!=NULL) mini = min(mini, abs(root->val - prev->val));
        prev = root;
        if(root->right!=NULL) func(root->right, prev);
    }
    int getMinimumDifference(TreeNode* root) {
        TreeNode* prev = nullptr;
        func(root, prev);
        return mini;
    }
};

File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int prev = -1;
    public int getMinimumDifference(TreeNode root) {
        return inorder(root);
    }

    public int inorder(TreeNode root){
        int ans = Integer.MAX_VALUE;

        if(root.left != null){
            ans = inorder(root.left);
        }

        if(prev != -1){
            ans = Math.min(root.val-prev,ans);
        }

        prev = root.val;

        if(root.right != null){
            ans = Math.min(ans,inorder(root.right));
        }

        return ans;

    }
}